File: Practice/src/coding_basic_training/ReturnValueDependingOnOddOrEven.java
package coding_basic_training;

public class ReturnValueDependingOnOddOrEven {
//	홀짝에 따라 다른 값 반환하기
	public int solution(int n) {
		int answer = 0;
		for (int i = n; i > 0; i = i - 2) {
			answer += n % 2 == 0 ? i * i : i;
		}
		return answer;
	}
}

File: Practice/src/coding_basic_training/ReturnValueDependingOnOddOrEven.js
function solution(n) {
    var answer = 0;
    var isEven = n % 2 == 0;
    for(let i = n ; i > 0 ; i-= 2){
        answer += isEven ? i*i : i; 
    }
    return answer;
    // return n % 2 == 0 ? n*(n+1)*(n+2)/6 : (n+1)/2*((n + 1)/2);
}